worker/upgradeseries/worker.go

```diff
@@ -92,6 +92,15 @@ type upgradeSeriesWorker struct {
 	machineStatus  model.UpgradeSeriesStatus
 	preparedUnits  []names.UnitTag
 	completedUnits []names.UnitTag
+
+	// This state is retained because we need to ensure that leaders are pinned
+	// only once if possible, on the first transition to UpgradeSeriesPrepare
+	// started. However there is nothing in the upgrade-series lock or workflow
+	// that denotes this as a discrete step.
+	// The only case where we will have multiple pin requests is if the worker
+	// restarts while still in the prepare-started state.
+	// This is not of too much concern, as the pin operations are idempotent.
```

Comment:
do we need to worry about this vs just calling Pin unconditionally?

Anyway, it is probably cleaner to just do it once. I would just hint that we probably don't need quite as much of a disclaimer here.

Reply:
If there are many units on a machine, the lock can get get a bit of churn while the units go through to _prepare-competed_, but I'll give the comment a haircut.